### DISCUSSION TITLE
how to use this disable button theme?

### DISCUSSION THREAD

**Discussion by yunchiri** [rydmike/flex_color_scheme#185] [Q&A] [answered]
<img width="1132" alt="image" src="https://github.com/rydmike/flex_color_scheme/assets/1159660/5f99a18b-8b13-4b5a-af72-aa32d2bfed6b">

how to use disbale "Elevated Button"

thanks library

**Comment by rydmike** [comment, marked as answer]
Hi @yunchiri, thank you for your question.

Can you explain a bit more about what you mean? Below a few guesses and their answers, but since I am not sure I understand the question correctly, the answer may not be what you are looking for.

**Q1**: How do I get this style of `ElevatedButton` in disabled mode in?
**A1**: Since you have it configured in the Playground, if you copy the theme from the playground generated code config, and use it in your app's `MaterialApp.theme` this is the style you will then get when the `ElevatedButton` is in its disabled state.

**Q2**: How do I recreate the shown style in the image?
**A2**: A bit hard to say exactly from the screen shot what configs have been made in the Playground. I would need a copy-paste of the generated code config, to tell you the exact steps are in the **Themes Playground** to get the shown button style.

**Q3**: How do I disable a button, in this case the `ElevatedButton` in Flutter?
**A3**: Not really a FlexColorScheme or Themes Playground question, but the buttons get their disabled style when their `onPressed` callback is set to null. This is just how Flutter Material buttons work.

So to disable a button conditionally you can e.g. use a ternary operator:

```dart
   :
   ElevatedButton(
     onPressed: buttonIsDisabled ? null : () { ...enabled press actions },
     child: Text('Elevated button'),
   ),
   :
```

Hope this helps, this does not seem to be an issue though, so I'm closing the issue and moving the question to the **Discussions** section as a Q&A. Feel free to continue the discussion there and provide more info if the above answer did not help.

**Comment by yunchiri** [comment]
Thanks for reply


<img width="358" alt="image" src="https://github.com/rydmike/flex_color_scheme/assets/1159660/9ae7b851-114a-4939-bb18-7db1c90eee2d">

I just want how to use code disabled button style using flex color scheme

I did copy and paste Theme code in main 
after ElevatedButton is good like example 
but I don't know how to using disabled style using flex color scheme
playground disabled style button is container()? or eleavtedButton with custom style?

could you show me code what  disabled button like [Filled button] style using flex color scheme

**Reply by rydmike** [reply]
Not sure I understand the question fully, but if you:

* Have added `flex_color_scheme` package to your `pubspec.yaml`
* Are using the theme you made with **Themes Playground** to your `theme` and `darkTheme` in your `MaterialApp`.

**Then** when you use the `ElevatedButton` and disable it in code, it will look like in the image automatically, it is just the same button that is in its disabled state.

Every component that can be disabled also have a disabled style that they get when you use whatever mechanism that the Flutter component in question uses, to enable its disabled state. This is not FlexColorScheme specific. In the Q&A 3 above I show how to disable the `ElevatedButton`, to the style it has then. Obviously in the example you have to defined the **bool** `isDisabled` to `true` to show the disabled state. If you just want to put a disabled button on the screen that cannot be used ever, well then just set its `onPressed` callback to `null`.

   :
   ElevatedButton(
     onPressed: null,
     child: Text('Elevated button'),
   ),
   :
  

- - -

**FlexColorScheme** is a fancy `ThemeData` factory with a lot of conveniences built-in to make theming easier and make pretty themes. It still only returns a `ThemeData` object and does not change how Flutter or its Material widgets work. 

**Themes Playground** is a visual config tool, to define a **FlexColorScheme** setup without knowing the API for it. The configured theme that you see, is what you get.

**Reply by yunchiri** [reply]
oh really thanks, finally I did it 

Happy day 
Thanks good library
